keanu-python/tests/keanu_assert.py

```diff
@@ -0,0 +1,44 @@
+"""
+Some utility functions to compare tensor-like things
+They might be Keanu tensors or Numpy arrays
+"""
+def tensors_equal(t1, t2):
+    if not shapes_equal(__get_shape(t1), __get_shape(t2)):
+        return False
+        print("Shapes don't match", __get_shape(t1), __get_shape(t2))
+        return False
+    it1 = __as_list(t1)
```

Comment:
Should we use generators here rather than lists?  I guess the tensors could get quite big in theory so not having to copy them would be good?

Reply:
Good call. You mean I should use generator expressions instead of list comprehensions inside the `__as_list` function, right?

Reply:
Yep exactly - sorry - should have been clearer!